Code2CollegeBankingProject/BankingUI/views.py

```diff
@@ -109,26 +107,32 @@ def forgotpassword(request):
         else:
             userinfo = {"response": "This username doesn't exist"}
             return render(request, "forgotpassword.html", userinfo)
-    return render(request, "forgotpassword.html")
+    return render(request,"forgotpassword.html")
 
 
 @login_required(redirect_field_name="index")
 def account(request):
     customer_transactions = Transactions.objects.filter(account=request.user.id)
     customer_amount = 0
-    for transactions in customer_transactions:
-        if transactions.transactiontype == "Deposit":
-            customer_amount += transactions.amount
-        if transactions.transactiontype == "Withdraw":
-            customer_amount -= transactions.amount
+    balance = []
```

Comment:
Ah! This is trickier than it seems right? If the code is doing this, then I see
![image](https://user-images.githubusercontent.com/20591884/166337248-a1b69d01-c25d-4538-992c-d9fe86eb707b.png)

Basically the list is being inserted into every row which is not what the user wants to see 🙃

Maybe... we can try adding the balance in each transaction of  `customer_transactions`

Something like:

```
    for transaction in customer_transactions:
        transaction.balance = balance
        if transaction.transactiontype == "Deposit":
            customer_amount += transaction.amount
            balance += transaction.amount
        if transaction.transactiontype == "Withdraw":
            customer_amount -= transaction.amount
            balance -= transaction.amount
        transaction.balance = balance

```

Then in `account.html` template:
```
        {% for transaction in transactions %}
        <tr>
          <td>{{transaction.date}}</td>
          <td>{{transaction.transactiontype}}</td>
          <td>{{transaction.amount}}</td>
          <td>{{transaction.balance}}</td>
        </tr>
        {% endfor %}
```

See what we're doing? We are giving each transaction an extra attribute `balance`

Reply:
Done, I didn't know you can just add a attribute like that